src/components/PDFView/PDFPasswordForm.js

```diff
@@ -0,0 +1,163 @@
+        }
+        this.setState({password});
+    }
+
+    validate() {
+        if (!_.isEmpty(this.state.password)) {
+            return true;
+        }
+        this.setState({
+            validationErrorText: this.props.translate('attachmentView.passwordRequired'),
+        });
+        return false;
+    }
+
+    validateAndNotifyPasswordBlur() {
+        this.validate();
+        this.props.onPasswordFieldFocused(false);
+    }
+
+    showForm() {
+        this.setState({shouldShowForm: true});
+    }
+
+    render() {
+        const containerStyle = this.props.isSmallScreenWidth
+            ? [styles.flex1, styles.w100]
+            : styles.pdfPasswordForm.wideScreenWidth;
+
+        return (
+            <>
+                {this.state.shouldShowForm ? (
+                    <ScrollView
+                        keyboardShouldPersistTaps="handled"
+                        style={containerStyle}
+                        contentContainerStyle={[styles.ph5, styles.flex1, styles.justifyContentCenter]}
+                    >
+                        <View style={styles.mb4}>
+                            <Text>
+                                {this.props.translate('attachmentView.pdfPasswordForm.formLabel')}
+                            </Text>
+                        </View>
+                        <TextInput
+                            label={this.props.translate('common.password')}
+                            autoComplete="off"
+                            autoCorrect={false}
+                            textContentType="password"
+                            onChangeText={this.updatePassword}
+                            returnKeyType="done"
+                            onSubmitEditing={this.submitPassword}
+                            errorText={this.state.validationErrorText}
+                            onFocus={() => this.props.onPasswordFieldFocused(true)}
+                            onBlur={this.validateAndNotifyPasswordBlur}
+                            autoFocus={this.props.shouldAutofocusPasswordField}
+                            secureTextEntry
+                        />
+                        {this.props.isPasswordInvalid && (
+                        <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt3]}>
+                            <Icon src={Expensicons.Exclamation} fill={colors.red} />
+                            <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
+                                <Text style={styles.mutedTextLabel}>
+                                    {this.props.translate('attachmentView.passwordIncorrect')}
+                                </Text>
+                            </View>
+                        </View>
+                        )}
+                        <Button
+                            text={this.props.translate('common.confirm')}
+                            onPress={this.submitPassword}
```

Comment:
Coming from https://github.com/Expensify/App/issues/25570

We should have ensured that the virtual keyboard stays open on mWeb when a wrong password is provided.